internal/mode/static/nginx/runtime/verify.go

```diff
@@ -19,15 +19,29 @@ const configVersionURI = "/var/run/nginx/nginx-config-version.sock"
 var noNewWorkersErrFmt = "reload unsuccessful: no new NGINX worker processes started for config version %d." +
 	" Please check the NGINX container logs for possible configuration issues: %w"
 
-// verifyClient is a client for verifying the config version.
-type verifyClient struct {
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . verifyClient
+
+type nginxConfigVerifier interface {
+	GetConfigVersion() (int, error)
+	WaitForCorrectVersion(
+		ctx context.Context,
+		expectedVersion int,
+		childProcFile string,
+		previousChildProcesses []byte,
+		readFile ReadFileFunc,
+	) error
+	EnsureConfigVersion(ctx context.Context, expectedVersion int) error
+}
+
+// VerifyClientImpl is a client for verifying the config version.
+type VerifyClientImpl struct {
```

Comment:
no need for Impl now:
```suggestion
type VerifyClient struct {
```

Reply:
okey